module/zfs/zil.c

```diff
@@ -1748,6 +1748,9 @@ zil_lwb_commit(zilog_t *zilog, itx_t *itx, lwb_t *lwb)
 			error = zilog->zl_get_data(itx->itx_private,
 			    itx->itx_gen, lrwb, dbuf, lwb,
 			    lwb->lwb_write_zio);
+			if (dbuf != NULL && error == 0)
+				bzero((char *)dbuf + lrwb->lr_length,
+				    dlen - lrwb->lr_length);
```

Comment:
I haven't touched this for a while, but I think this may be wrong.  dlen is a total data size before possible splitting, while lrwb->lr_length is what to be written in current log block.  If there is more than one block, you may bzero() more than you should.

Reply:
Thanks, indeed, this triggers a panic in the test suite. I'm testing an updated version now.